src/app/components/elements/Carousel.tsx

```diff
@@ -88,28 +89,49 @@ const ControlledCarouselInstance = ({ children, collectionTag }: any) => {
 };
 
 export const ResponsiveCarousel = ({ groupingLimit, children, collectionTag = "div", className }: any) => {
-  const tuple: any = [];
+  const [screenWidth, setScreenWidth] = useState(window.innerWidth);
 
-  if (!groupingLimit || groupingLimit == 0) {
-    groupingLimit = 3;
+  useEffect(() => {
+    const handleResize = () => setScreenWidth(window.innerWidth);
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  let effectiveGroupingLimit = groupingLimit || GROUP_LIMITS.DEFAULT;
+
+  if (screenWidth >= SCREEN_SIZES.MOBILE && screenWidth <= SCREEN_SIZES.TABLET) {
+    effectiveGroupingLimit = GROUP_LIMITS.TABLET;
+  } else if (screenWidth < SCREEN_SIZES.MOBILE) {
+    effectiveGroupingLimit = GROUP_LIMITS.MOBILE;
   }
 
-  children.forEach((child: any, index: number) => {
-    if (index % groupingLimit === 0) {
-      tuple.push([]);
-    }
-    tuple[Math.floor(index / groupingLimit)].push(child);
-  });
+  type TupleItem = React.ReactNode[];
+  const [tuple, setTuple] = useState<TupleItem[]>([]);
```

Comment:
is this type right? Just to double check it's a nested array?

Reply:
You're not wrong, It's a nested array 👀 I'm going to guess that there is an even better way to improve this 😅

Reply:
Actually, it's good - I just wanted to double check :)